src/views/Generator/RuleEditor/Typeahead/Typeahead.tsx

```diff
@@ -0,0 +1,55 @@
+import { Box, Card, TextField } from '@radix-ui/themes'
+import { forwardRef, ComponentProps } from 'react'
+
+import { SuggestionMode, useTypeahead } from './useTypeahead'
+
+export type TypeaheadProps = ComponentProps<typeof TextField.Root> & {
+  options: string[]
+  mode: SuggestionMode
+}
+
+export const Typeahead = forwardRef<HTMLInputElement, TypeaheadProps>(
```

Comment:
I noticed some styling issues when scrolling a long list of suggestions.

<img width="478" height="209" alt="Image" src="https://github.com/user-attachments/assets/b12b229b-dc4b-4408-b7e6-0fa72a15fe3e" />